### PR TITLE
[SE-0382] Stage in prototypes of macros defined in the standard library.

### PR DESCRIPTION
**Pull request by DougGregor**
These are inactive at the moment; we'll conditionally take advantage of them later on. Part of rdar://104346457.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: stdlib/public/core/Macros.swift [hunk 1]
```diff
@@ -24,4 +23,40 @@
 public macro externalMacro<T>(module: String, type: String) -> T =
     Builtin.ExternalMacro
```
Comment (rxwei) [inline stdlib/public/core/Macros.swift:24]: Trivial nit: Inconsistent indentation with the rest

```suggestion
  Builtin.ExternalMacro
```

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Pull request merged**